UI: Add retina support and updated icons

DungFu opened the pull request:
OBS Studio currently does not support retina rendering for any of the
images in the app. This adds support for retina pixmap rendering as
well as adding higher resolution versions of each icon that was not
a high enough resolution to support 2x. The icons work when switching
themes and will only render the 2x versions when the device pixel
ratio is greater than or equal to 2.

I also added credits to the readme for images that were already being
used that require credit to the author. If the OBS community has
paid for these images already, I can remove the credits from the
readme. The credit is for the invisible, visible, and gear icons.

## Before
![obs-non-retina](https://user-images.githubusercontent.com/443370/44503364-cfdda400-a64a-11e8-8924-03a2bfe78f83.png)

## After
![obs-retina](https://user-images.githubusercontent.com/443370/44942841-9e15bd00-ad6f-11e8-85b8-5004c3bfad66.png)

RytoEX commented on UI/locked-checkbox.cpp:
```diff
@@ -7,10 +7,17 @@
-		QPixmap::fromImage(QImage(":/res/images/unlocked_mask.png"));
+	if (devicePixelRatio() >= 2) {
+		lockedImage =
+			QPixmap::fromImage(QImage(":/res/images/locked_mask@2x.png"));
```
Style:  These lines are longer than 80-characters.  You will probably have to line break them differently, like so:
```
		lockedImage = QPixmap::fromImage(QImage(
				":/res/images/locked_mask@2x.png"));
```

DungFu replied:
Will fix

RytoEX commented on UI/locked-checkbox.cpp:
```diff
@@ -31,6 +38,6 @@ void LockedCheckBox::paintEvent(QPaintEvent *event)
 
 	QPainter p(this);
-	p.drawPixmap(0, 0, image.width(), image.height(),
+	p.drawPixmap(0, 0, 16, 16,
```
Why are these values being hardcoded here?

RytoEX replied:
How does this affect the icon displays on other operating systems, if at all?

RytoEX commented on UI/visibility-checkbox.cpp:
```diff
@@ -7,10 +7,17 @@
-		QPixmap::fromImage(QImage(":/res/images/invisible_mask.png"));
+	if (devicePixelRatio() >= 2) {
+		checkedImage =
+			QPixmap::fromImage(QImage(":/res/images/visible_mask@2x.png"));
```
Style: These lines are longer than 80-characters. You will probably have to line break them differently, like so:
```
		checkedImage = QPixmap::fromImage(QImage(
				":/res/images/visible_mask@2x.png"));
```

DungFu replied:
Will fix

RytoEX commented on UI/visibility-checkbox.cpp:
```diff
@@ -31,6 +38,6 @@ void VisibilityCheckBox::paintEvent(QPaintEvent *event)
 
 	QPainter p(this);
-	p.drawPixmap(0, 0, image.width(), image.height(),
+	p.drawPixmap(0, 0, 16, 16,
```
Why are these values hardcoded?

DungFu commented on README.rst:
```diff
@@ -47,3 +47,7 @@ Contributing
+
+Credits
+-------
+ - Icons made by [Freepik](http://www.freepik.com) from [www.flaticon.com](https://www.flaticon.com/) is licensed by [CC 3.0 BY](http://creativecommons.org/licenses/by/3.0/)
```
Will fix

DungFu commented:
@DDRBoxman I could not find a higher resolution source for that icon and figured that if the icon was to be changed, that it should probably happen in a separate PR.

If anyone has the original source for that icon, that would be great and I can add it to this PR.

DungFu commented:
@DDRBoxman I managed to find the source for the output icon and updated the commit with the new file. The screenshots are also updated to reflect the new image.

RytoEX commented:
@DungFu 
Following up on this, could you please provide links to the icon pack or individual icon pages on Flaticon and/or Freepik's site?  That way we can refer back to it should the need arise.
